initial integration PR for ufdr-mounter plugin

SribatschaMaharana opened the pull request:
This plugin mounts UFDR forensic archives as a FUSE-based virtual file system for inspection via the RescueBox interface.

Libraries used, probably:
fusepy – ISC License https://github.com/fusepy/fusepy/blob/master/LICENSE


No external models are used in this plugin. The plugin supports reading UFDR file archives using FUSE and will be connected to Rescuebox.

prasannals commented:
Thanks for the PR. You will have to remove Flask-ML once you migrate to RescueBox. Can you please create a folder named "LICENSES" within your plugins root directory and add the license files for your dependencies in that folder?

Example from the deep fake detection project: https://github.com/lous-e/RescueBox/tree/test/src/deepfake-detection/LICENSES

SribatschaMaharana commented:
Most recent commit changes made:
refactored flask_ml code to RB library functions. 
updated code according to some copilot suggestions
added license folder for fuse

remaining work: adding tests

Available commands:
RescueBox % poetry run python src/ufdr_mounter/ufdr_mounter/ufdr_server.py --help                                                                        
╭─ Commands 
│ /ufdr_mounter/api/app_metadata                                                                                                                                                       
│ /ufdr_mounter/api/routes               Lists all the routes/endpoints available in the Flask app.                                                                                         
│ /ufdr_mounter/mount                                                                                                                                                                  
│ /ufdr_mounter/mount/task_schema     

run mount command example:
% /poetry run python src/ufdr_mounter/ufdr_mounter/ufdr_server.py /ufdr_mounter/mount --"src/ufdr_mounter/ufdr_mounter/testdata/test.ufdr,./mnt/test_mount " " "

nb950 commented:
merged to https://github.com/UMass-Rescue/RescueBox/tree/plugins_from_spring_2025
